packages/compose/src/hooks/use-resize-observer/index.native.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { View, StyleSheet } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Hook which allows to listen the resize event of any target element when it changes sizes.
+ *
+ * @return {Object} Measurements object with properties `width` and `height`.
+ *
+ * @example
+ *
+ * ```js
+ * const App = () => {
+ * 	const [ resizeListener, sizes ] = useResizeObserver();
+ *
+ * 	return (
+ * 		<View>
+ * 			{ resizeListener }
+ * 			Your content here
+ * 		</View>
+ * 	);
+ * };
+ * ```
+ *
+ */
+const useResizeObserver = () => {
+	const [ measurements, setMeasurements ] = useState( null );
+
+	const onLayout = useCallback( ( { nativeEvent } ) => {
+		const { width, height } = nativeEvent.layout;
+
+		setMeasurements( { width, height } );
+	}, [] );
+
+	const resizeObserver = () => {
+		return (
+			<View
+				style={ { ...StyleSheet.absoluteFill } }
+				onLayout={ onLayout }
+			/>
+		);
+	};
+
+	return [ resizeObserver(), measurements ];
```

Comment:
Does the function provide much usefulness, vs. just inlining the `View` element?

```suggestion
	return [
		<View { /* ... */ } />,
		measurements,
	];
```

Reply:
Will change it!